### PR TITLE
VM groups

### PR DESCRIPTION
**Pull request by atsikham**
* All resources are in the single resource group
* Default disk configuration is used, there is a separate tasks related to disks - https://github.com/epiphany-platform/m-azure-basic-infrastructure/issues/42 and https://github.com/epiphany-platform/m-azure-basic-infrastructure/issues/28
* Default security group to allow public SSH connections is used, if there are requirements exist to have configurable NSGs or different NSGs per VM group, a separate task should be created for that
* Not tested, but needs to be checked that after the Golang changes are applied we still have working examples (basic_flow)  in Azure KS module. @mkyc could you comment that?

Related e-structures PR: https://github.com/epiphany-platform/e-structures/pull/2

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,8 +3,41 @@
...
+
+# Crash log files
+crash.log
+
+# Exclude all .tfvars files, which are likely to contain sentitive data, such as
+# password, private keys, and other secrets. These should not be part of version
+# control as they are data points which are potentially sensitive and subject
+# to change depending on the environment.
+#
+*.tfvars
```
Comment (mkyc) [inline .gitignore:25]: Shouldn't *.tfvars.json also be here?

File: resources/terraform/main.tf [hunk 2]
```diff
@@ -3,17 +3,18 @@ resource "azurerm_resource_group" "rg" {
...
 
-  instances     = var.vms_count
-  name          = var.name
-  rg_name       = azurerm_resource_group.rg.name
-  vnet_id       = azurerm_virtual_network.vnet.id
-  location      = var.location
-  service       = azurerm_subnet.subnets[0].name
-  use_public_ip = var.use_public_ip
-  subnet_id     = azurerm_subnet.subnets[0].id
+  name              = var.name
```
Comment (mkyc) [inline resources/terraform/main.tf:10]: Shouldn't it be rather `var.vm_groups[count.index].name` ?

Comment (atsikham) [reply resources/terraform/main.tf:10]: Resource names include related VM group's name with this prefix, check `vm_group` module, `main.tf`

**Review by mkyc — APPROVED** [review]
👍

**Review by erzetpe — APPROVED** [review]

**Pull request merged**